### PR TITLE
More cowbell

### PR DESCRIPTION
**Pull request by olydis**
- directives for renaming and removing operations, types and properties
- tests for that
- CWD voodoo in the bootstrapper

### CODE REVIEW

**Comment by olydis** [comment]
# 🤖 AutoRest automatic publish job 🤖
~~~ Haskell
> will publish once PR gets merged
~~~

File: src/autorest-core/resources/default-configuration.md [hunk 1]
```diff
@@ -459,4 +469,37 @@ declare-directive:
...
+      "where-operation": $.from,
+      transform: `$.operationId = ${JSON.stringify($.to)}`
+    }
+  remove-type: >-
+    {
+      from: 'swagger-document',
+      "where-type": $,
+      transform: 'return undefined'
+    }
+  rename-type: >-
```
Comment (fearthecowboy) [inline src/autorest-core/resources/default-configuration.md]: model

File: src/autorest-core/app.ts [hunk 2]
```diff
@@ -285,7 +285,18 @@ async function currentMain(autorestArgs: string[]): Promise<number> {
   subscribeMessages(api, () => exitcode++);
   const artifacts: Artifact[] = [];
   const clearFolders: string[] = [];
-  api.GeneratedFile.Subscribe((_, artifact) => artifacts.push(artifact));
+  api.GeneratedFile.Subscribe((_, artifact) => {
+    if (artifact.uri.startsWith("stdout://")) {
+      config.Message({
```
Comment (olydis) [inline src/autorest-core/app.ts]: note to tomorrow Johannes: why didn't you just put that in the emitter, as the message even suggests?

File: src/autorest-core/lib/pipeline/artifact-emitter.ts [hunk 3]
```diff
@@ -16,18 +17,30 @@ function IsOutputArtifactOrMapRequested(config: ConfigurationView, artifactType:
 
 async function EmitArtifactInternal(config: ConfigurationView, artifactType: string, uri: string, handle: DataHandle): Promise<void> {
   config.Message({ Channel: Channel.Debug, Text: `Emitting '${artifactType}' at ${uri}` });
+  const emitArtifact = (artifact: Artifact): void => {
+    if (artifact.uri.startsWith("stdout://")) {
```
Comment (fearthecowboy) [inline src/autorest-core/lib/pipeline/artifact-emitter.ts]: So, in AaaL, the consumer sets the `output-folder` to `stdout://` and they get all the files via messages?

File: src/autorest-core/package.json [hunk 4]
```diff
@@ -62,7 +62,7 @@
       "safe-eval": "^0.3.0",
       "source-map": "^0.5.6",
       "strip-bom": "^3.0.0",
-      "swagger2openapi": "olydis/swagger2openapi#b9f1e7337a8fad741ad5976b1502357a67818ec5",
+      "swagger2openapi": "olydis/swagger2openapi#15da1ddb9e3b56891991788784f546967cba3f52",
```
Comment (olydis) [inline src/autorest-core/package.json]: rel: https://github.com/Azure/autorest/issues/2758

**Pull request merged**